src/main/java/de/tum/in/www1/artemis/service/UserService.java

```diff
@@ -146,8 +149,15 @@ public StandardPBEStringEncryptor encryptor() {
         return encryptor;
     }
 
-    @Value("${artemis.encryption-password}")
-    private String ENCRYPTION_PASSWORD;
+    /**
+     * Returns the password of the user as a char array (as is the Java convention for handling sensitive Strings)
+     *
+     * @param user The user for whom to fetch the password
+     * @return The password of the specified user
+     */
+    public char[] getPasswordForUser(User user) {
```

Comment:
please remove and instead use `decryptPasswordByLogin`

Reply:
I kept this method and renamed it, because the `decryptPasswordByLogin` method also fetches the user from the database, while we already have the user in this case. The `decryptPasswordByLogin` method now calls this method and also returns a (Optional) char array.